components/brave_new_tab_ui/components/default/sponsoredImage/sponsosredImageClickArea.tsx

```diff
@@ -0,0 +1,41 @@
+import { OpenNewIcon } from 'brave-ui/components/icons'
+import * as React from 'react'
+import styled from 'styled-components'
+
+const Indicator = styled.span`
+  display: none;
+
+  position: absolute;
+  top: 8px;
+  right: 8px;
+
+  width: 20px;
+  height: 20px;
+  color: white;
+`
+
+const Link = styled.a`
+  position: absolute;
+  top: 8px;
+  left: 8px;
+  right: 56px;
+  bottom: 8px;
+  display: block;
+
+  &:hover {
+    cursor: pointer;
+    ${Indicator} {
+      display: block;
+    }
+  }
+`
+
+export default function SponsoredImageClickArea(props: { sponsoredImageUrl: string, onClick: () => void }) {
+  return (
+    <Link href={props.sponsoredImageUrl} rel="noreferrer noopener" onClick={props.onClick}>
```

Comment:
@fallaciousreasoning can we clean up the link to only accept `https?://` schemes?

Reply:
We're already using this link on the page, and as @fmarier mentioned, the source of the URLs are trusted, so I don't know if it's worth adding that check here.

Maybe, in future, we could introduce a `Link` component which does a bunch of security checks on the URL? Maybe the one in Leo would be a good target for this, as then we'd get the benefits everywhere Leo is used?

Could be worth filing an issue on https://github.com/brave/leo/issues/new?

Reply:
Actually @simonhong, is it okay that I have the `noreferrer` here? How do sponsored sites track clicks?

Reply:
@fallaciousreasoning why an issue in Leo and not in `brave-browser`?
It seems like a good practice to just use a `SafeLink`^(possible name) component in `brave-core`

Reply:
I think, ideally, that component should exist in Leo, so we can use it in other Brave projects, if the need arises

Reply:
> Actually @simonhong, is it okay that I have the `noreferrer` here? How do sponsored sites track clicks?

I think `noreferrer` is fine. When `onClickLogo()` is clicked, it notifies to browser with current sponsored images's metadata. BTW, we are using `onClickLogo()` for notifying about user's logo clicking.
Not sure, this could be used here. @tmancey @aseren

Reply:
If we don't have to distinguish between the two(logo click and image area click), I think it's fine. @tmancey @aseren WDYT?

Reply:
As we do the same as for the Logo click (i.e. open the same URL), I think it is fine to call `onClickLogo()` when an image area is clicked. 
Should we also add `rel="noreferrer noopener"` to the Logo click anchor?